File: AlkmaarseRestaurants/Models/Restaurant/RestaurantModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AlkmaarseRestaurants.Models.Restaurant
{
    public class RestaurantModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Address { get; set; }
        public string StreetName { get; set; }
        public string StreetNumber { get; set; }
        public string PostalCode { get; set; }
        public string ImageLogo { get; set; }
        public RestaurantCategoryModel RestaurantType { get; set; }
        public RestaurantInfoModel RestaurantInfo { get; set; }
        public List<RestaurantMenuItemModel> RestaurantMenu { get; set; }
    }
}

File: AlkmaarseRestaurants/angular/src/app/shared/models/Restaurant/RestaurantModel.ts
import { IRestaurantModel } from "./IRestaurantModel";
import { RestaurantInfoModel } from "./RestaurantInfoModel";
import { RestaurantMenuItemModel } from "./RestaurantMenuItemModel";
import { RestaurantCategoryModel } from "./RestaurantCategoryModel";

export class RestaurantModel implements IRestaurantModel {
    public id: number;
    public name: string;
    public address: string;
    public streetName: string;
    public streetNumber: string;
    public postalCode: string;
    public imageLogo: string;
    public restaurantType: RestaurantCategoryModel;
    public restaurantInfo: RestaurantInfoModel;
    public restaurantMenu: Array<RestaurantMenuItemModel>;

    constructor() {
        this.id = 0;
        this.name = "";
        this.address = "";
        this.streetName = "";
        this.streetNumber = "";
        this.postalCode = "";
        this.imageLogo = "";
        this.restaurantType = new RestaurantCategoryModel();
        this.restaurantInfo = new RestaurantInfoModel();
        this.restaurantMenu = [];
    }
}